### PR TITLE
Docs: Add server upgrade instructions.

### PR DESCRIPTION
**Pull request by simitt**
implements elastic/apm-server#1008

related to https://github.com/elastic/stack-docs/pull/81

@dedemorton how can I add APM Server to https://www.elastic.co/products/upgrade_guide?

update:
screen shot 2018-07-17 at 16 36 49

### CODE REVIEW

**Comment by dedemorton** [comment]
@simitt 

@debadair is updating the guide right now (I think she already has it on staging). I'll ping her to let her know that you've asked, and you can follow up with her next week.

**Comment by debadair** [comment]
@simitt I've opened a separate issue for adding APM server to the interactive guide: https://github.com/elastic/stack-docs/issues/83.

It's a bit fragile to update, so I want to tackle that separately from the fixes I currently have staged for release.

File: docs/upgrading.asciidoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[[upgrading]]
+=== Upgrading APM Server
```
Comment (dedemorton) [inline docs/upgrading.asciidoc]: Have you thought about putting the upgrade content at the top-level in the TOC (to be consistent with other docs in the stack)? It's fine to do it this way as long as you think users will see the content.

File: docs/upgrading.asciidoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+When you have ensured everything is properly configured and updated,
+start the APM Server again.
+
+By default index names and the mapping templates are versioned. 
+When starting the new installation of APM Server new indices with the current version are created, 
+and the correct template for this version is applied automatically. 
+
+[[breaking-changes]]
+[float]
+==== Breaking Changes
```
[the remark this answers is not in the document]
Comment (simitt) [reply docs/upgrading.asciidoc]: So far this upgrade guide only considers minor upgrades, as we haven't had a major upgrade.
Within minor upgrades we aim for having no breaking changes, that's why I put the section here and not on a separate page.

[the remark this answers is not in the document]
Comment (simitt) [reply docs/upgrading.asciidoc]: I gave it a second thought and now put _Upgrading_ top level and _Breaking changes_ within the upgrade guide but as a separate page. WDYT?

File: docs/upgrading.asciidoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+[float]
+==== 6.3
+no breaking changes in APM Server
+
+https://www.elastic.co/guide/en/beats/libbeat/current/breaking-changes-6.3.html[breaking changes in libbeat]
+
+[float]
+==== 6.2
+First stable version.
```
Comment (jalvz) [inline docs/upgrading.asciidoc]: what "stable" means? (GA?)

File: docs/upgrading.asciidoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+When everything is properly configured and updated, start the APM server.
+
+When you start the APM server after upgrading, it creates new indices that use the current version,
+and applies the correct template automatically.
+
+[[breaking-changes]]
+=== Breaking Changes
+APM Server is built on top of {beats-ref}/index.html[libbeat].
+As such any breaking change in libbeat is also considered to be a breaking change in APM Server.
+
```
Comment (jalvz) [inline docs/upgrading.asciidoc]: This is OK, but separately from this we could discuss/create a policy around what is a breaking change and document it. I think would be useful for corner cases like rephrase/removing logs, etc.

**Pull request merged**